### PR TITLE
Moved the test classes to the test directory

### PR DESCRIPTION
**Pull request by etorreborre**
@markhibberd I have tested the publishing of tests jars and the `test->test` dependency on a `saws` branch and things seem to work [fine](https://github.com/ambiata/saws/pull/93).

### CODE REVIEW

File: project/build.scala [hunk 1]
```diff
@@ -140,14 +140,14 @@ object build extends Build {
     , settings = standardSettings ++ lib("testing") ++ Seq[Settings](
       name := "mundane-testing-extra"
     ) ++ Seq[Settings](libraryDependencies ++= depend.specs2 ++ depend.specs2Extra)
-  ).dependsOn(testing)
+  ).dependsOn(testing % "test->test")
```
Comment (charleso) [inline project/build.scala]: Not that it matters, but that doesn't look right. Is `testing-extra` supposed to extend `testing` with more util-like functionality (and depend on it for `compile`).

In any case it doesn't look like `testing-extra` actually needs `testing` in the first place

Comment (etorreborre) [inline project/build.scala]: Yes I removed it.

**Comment by etorreborre** [comment]
@markhibberd are ok with this?

**Comment by etorreborre** [comment]
@markhibberd @charleso I need to go forward with this to remove properly test dependencies from ivory.

**Comment by markhibberd** [comment]
@etorreborre I am fine with this change, but it isn't required to do the right thing in the other projects. There is already an approach that has always been used for those projects. At some point, something has incorrectly added a dependency, and that should be fixed independently of this change IMO.

**Comment by etorreborre** [comment]
@markhibberd my comment is wrong. The specs2 dependency is already out of ivory.

**Pull request merged**